content/200-concepts/100-components/02-prisma-client/052-extend-client.mdx

```diff
@@ -0,0 +1,400 @@
+---
+title: 'Validation, Computed Fields, and Custom Logic'
```

Comment:
Not too sure about the organization. I feel like this fits more in Concepts, but unfortunately, I wrote this like a Guide.  Reviewing the website, I'm a bit worried that putting it in Guide's under a generic title like "How to Extend the Prisma Client" will make it undiscoverable.

I'm happy to slice and dice the content to make it fit more into Concepts. Otherwise, we call it a Guide with a better title like "How to add runtime validation, computed fields and custom logic to your app". This could even be a couple guides.

Reply:
@matthewmueller and @janpio this is indeed a legit challenge!

The restructuring I am looking at (too slowly!) tries to answer this problem, with a version of what Jan suggests. The Concepts section is for high-level concepts, the WHY and the WHAT. Then under the Guides is the HOW content, the procedural steps. I haven't looked closely at this one yet, but so far it seems we could have most of this in a new guide (under Guides) and then under concepts, perhaps on [this page](https://www.prisma.io/docs/concepts/components/prisma-client), we link to the guide about these advanced steps of working with the Client.

Reply:
@matthewmueller another thought... what do you think of separating this PR into two PRs... that way we can go ahead and merge the ones we know are good to go (like updating the Preview features lists), and have the new file (052-extend-client.mdx) in a separate PR where we can work on our questions about it?

Reply:
> (like updating the Preview features lists)

Oh shoot. This is a git mistake. This happened to me recently before. I pulled in other people's commits. I'll clean that up. This PR should only be about the Extending Prisma Client Guide. 

I plan to add 1 (or more) other pages in Concepts like Validation, Computed Fields, Excluding Fields and Custom Logic that points to sections of the Guide. And then eventually we can replace each of those links with something that Prisma natively does.